cooperative_lanechange/src/cooperative_lanechange.cpp

```diff
@@ -0,0 +1,897 @@
+                veh2_speed = rwol[0].object.velocity.twist.linear.x;
+                foundRoadwayObject = true;
+                break;
+            }
+        }
+        if(foundRoadwayObject){
+            //get current_gap
+            double current_gap = find_current_gap(veh2_lanelet_id,veh2_downtrack);
+            //get desired gap - desired time gap (default 3s)* relative velocity
+            double relative_velocity = current_speed_ - veh2_speed;
+            double desired_gap = desired_time_gap_ * relative_velocity;      
+            
+            if(current_gap > desired_gap){
+                negotiate = false;  //No need for negotiation
+            }
+            
+        }
+        else{
+            ROS_WARN_STREAM("Did not find a connected and automated vehicle roadway object");
+        }
+
+        //plan lanechange without filling in response
+        std::vector<cav_msgs::TrajectoryPlanPoint> planned_trajectory_points = plan_lanechange(req);
+        
+        ros::Time request_sent_time;
+
+        if(negotiate){
+            //send mobility request
+            //Planning for first lane change maneuver
+            cav_msgs::MobilityRequest request = create_mobility_request(planned_trajectory_points, maneuver_plan[0]);
+            outgoing_mobility_request_.publish(request);
+            request_sent_time = ros::Time::now();
+            cav_msgs::LaneChangeStatus lc_status_msg;
+            lc_status_msg.status = cav_msgs::LaneChangeStatus::PLAN_SENT;
+            lc_status_msg.description = "Requested lane merge";
+            lanechange_status_pub_.publish(lc_status_msg);
+        }
+
+        //if ack mobility response, send lanechange response
+        if(!negotiate || is_lanechange_accepted_){
+            cav_msgs::TrajectoryPlan trajectory_plan;
+            trajectory_plan.header.frame_id = "map";
+            trajectory_plan.header.stamp = ros::Time::now();
+            trajectory_plan.trajectory_id = boost::uuids::to_string(boost::uuids::random_generator()());
+
+            trajectory_plan.trajectory_points = planned_trajectory_points;
+            trajectory_plan.initial_longitudinal_velocity = std::max(req.vehicle_state.longitudinal_vel, minimum_speed_);
+            resp.trajectory_plan = trajectory_plan;
+
+            for (int i=0; i<req.maneuver_plan.maneuvers.size(); i++){
+                if (req.maneuver_plan.maneuvers[i].type == cav_msgs::Maneuver::LANE_CHANGE){
+                    resp.related_maneuvers.push_back(i);
+                    break;
+                }
+            }
+            resp.maneuver_status.push_back(cav_srvs::PlanTrajectory::Response::MANEUVER_IN_PROGRESS);
+
+        }
+        else{
+            if(!negotiate){
+                request_sent_time = ros::Time::now();
+            }
+            ros::Time planning_end_time = ros::Time::now();
+            ros::Duration passed_time = planning_end_time - request_sent_time;
```

Comment:
`request_sent_time` is allocated on the stack in line 245 right? So here we're just measuring the time it takes us to navigate from line 245 to line 285? Is the intent here to track this time across multiple iterations of this callback?

Reply:
Does it make more sense to track the request state and timing as a class member variable to ensure that the data persists across multiple calls to this callback?